modules/apps/captcha/captcha-taglib/src/main/resources/META-INF/resources/captcha/simplecaptcha.jsp

```diff
@@ -65,23 +68,44 @@ String url = (String)request.getAttribute("liferay-captcha:captcha:url");
 
 			if (refreshCaptcha && !hasEventAttached) {
 				hasEventAttached = true;
-				refreshCaptcha.addEventListener('click', () => {
-					var url = Liferay.Util.addParams(
-						't=' + Date.now(),
-						'<%= HtmlUtil.escapeJS(url) %>'
-					);
-
-					var captcha = document.getElementById(
-						'<portlet:namespace />captcha'
-					);
-
-					if (captcha) {
-						captcha.setAttribute('src', url);
-					}
-				});
+				refreshCaptcha.addEventListener('click', handleCaptcha);
 			}
 		}
 
+		function handleCaptcha() {
+			if (<%= url.startsWith("/o/") %>) {
+				return getCaptchaSourceImage('<%= url %>');
```

Comment:
make it safe <%= HtmlUtil.escapeJS(url) %>

Reply:
Fixed it. Thanks